src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java

```diff
@@ -0,0 +1,243 @@
+
+        String receivedPath = request.get(dragItemPath, HttpStatus.OK, String.class);
+        assertThat(receivedPath).isEqualTo("some data");
+    }
+
+    @Test
+    @WithMockUser(value = "instructor1", roles = "INSTRUCTOR")
+    public void testGetFileUploadSubmission() throws Exception {
+        database.addCourseWithTwoFileUploadExercise();
+        FileUploadExercise fileUploadExercise = (FileUploadExercise) exerciseRepo.findAll().get(0);
+        FileUploadSubmission fileUploadSubmission = ModelFactory.generateFileUploadSubmission(true);
+        fileUploadSubmission = database.addFileUploadSubmission(fileUploadExercise, fileUploadSubmission, "student1");
+
+        MockMultipartFile file = new MockMultipartFile("file", "file.png", "application/json", "some data".getBytes());
+        JsonNode response = request.postWithMultipartFile("/api/fileUpload?keepFileName=true", file.getOriginalFilename(), "file", file, JsonNode.class, HttpStatus.CREATED);
+        String responsePath = response.get("path").asText();
+        String filePath = fileService.manageFilesForUpdatedFilePath(null, responsePath,
+                fileUploadSubmission.buildFilePath(fileUploadExercise.getId(), fileUploadSubmission.getId()), fileUploadSubmission.getId(), true);
+
+        fileUploadSubmission.setFilePath(filePath);
+
+        // get access token
+        String accessToken = request.get("/api/files/attachments/access-token/file.png", HttpStatus.OK, String.class);
+
+        String receivedFile = request.get(fileUploadSubmission.getFilePath() + "?access_token=" + accessToken, HttpStatus.OK, String.class);
+        assertThat(receivedFile).isEqualTo("some data");
+    }
+
+    @Test
+    @WithMockUser(value = "instructor1", roles = "INSTRUCTOR")
+    public void testGetLectureAttachment() throws Exception {
+        Lecture lecture = database.createCourseWithLecture(true);
+
+        Attachment attachment = ModelFactory.generateAttachment(ZonedDateTime.now(), lecture);
+
+        MockMultipartFile file = new MockMultipartFile("file", "attachment.png", "application/json", "some data".getBytes());
+        JsonNode response = request.postWithMultipartFile("/api/fileUpload?keepFileName=true", file.getOriginalFilename(), "file", file, JsonNode.class, HttpStatus.CREATED);
+        String responsePath = response.get("path").asText();
+        String attachmentPath = fileService.manageFilesForUpdatedFilePath(null, responsePath, Constants.LECTURE_ATTACHMENT_FILEPATH, lecture.getId(), true);
+
+        attachment.setLink(attachmentPath);
+        lecture.addAttachments(attachment);
+        lectureRepo.save(lecture);
+        attachmentRepo.save(attachment);
+
+        // get access token
+        String accessToken = request.get("/api/files/attachments/access-token/attachment.png", HttpStatus.OK, String.class);
+
+        String receivedAttachment = request.get(attachmentPath + "?access_token=" + accessToken, HttpStatus.OK, String.class);
```

Comment:
This GET request failes, because the attachment file cannot be found. I created and uploaded the file and moved it to the correct folder. What might I be missing here?

Reply:
The error 404 is thrown when the file should be returned, but it can not be found at the given path `attachmentPath`, allthough I saved it there just before.

Reply:
I committed the change that fixes this issue. You didn't put the lecture in the `id` folder, so it couldn't find it